rootclock.c

```diff
@@ -66,27 +87,67 @@ static void render_all(Drw *drw, Fnt *tf, Fnt *df, int show_date_flag,
                        Clr *bg_scm, Clr *time_scm, Clr *date_scm,
                        const char *time_fmt_s, const char *date_fmt_s,
                        int block_y_off_s, int line_spacing_s) {
+  char tbuf[64], dbuf[128];
   time_t now = time(NULL);
+  
+  if (now == (time_t)-1) {
+    /* time() failed, use fallback */
+    strcpy(tbuf, "--:--");
+    if (show_date_flag)
+      strcpy(dbuf, "Unknown Date");
+    goto skip_time_formatting;
+  }
+  
   struct tm *tm_info = localtime(&now);
-  char tbuf[64], dbuf[128];
-  strftime(tbuf, sizeof tbuf, time_fmt_s, tm_info);
-  if (show_date_flag)
-    strftime(dbuf, sizeof dbuf, date_fmt_s, tm_info);
+  if (!tm_info) {
+    /* localtime() failed, use fallback */
+    strcpy(tbuf, "--:--");
```

Comment:
The format of time could be different, for example `config.def.h` defines it with a ratio unicode symbol instead of a colon.